### PR TITLE
Remove Box2D dependency

### PR DESCRIPTION
**Pull request by spydon**
# Description

Remove Box2D as a dependency of Flame.
It is based on the vector2d-unification2 branch, so that needs to be merged first.

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] The continuous integration (CI) is passing

### CODE REVIEW

**Review by erickzanardo — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ The goal of this project is to provide a complete set of out-of-the-way solution
 Currently it provides you with:
  - a game loop
  - a component/object system
- - bundles a physics engine (box2d)
+ - a physics engine (box2d)
```
Comment (luanpotter) [inline README.md:22]: `(box2d, available through flame_box2d)`

File: doc/box2d.md [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # Box2D
 
-Although Flame does not implement Box2d itself, it bundles a forked port of the Java Box2d to Dart by Google.
+Although Flame does not fully implement Box2d itself, it bundles a forked and refactored port of the Java Box2d to Dart by Google.
```
Comment (luanpotter) [inline doc/box2d.md:3]: `it bundles` => `we maintain`

File: doc/box2d.md [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
 
-The source of the bundled box2d on Flame can be found [here](https://github.com/flame-engine/box2d.dart).
+The source of the box2d version that Flame can use can be found [here](https://github.com/flame-engine/box2d.dart).
 
 To use Box2D in your game you should add flame_box2d to your pubspec.yaml, which can be seen in the examples.
```
Comment (luanpotter) [inline doc/box2d.md:7]: I think we can spend an extra sentence here clearly making the distinction between:
* our fork of box2d
* flame_box2d

**Review by luanpotter — APPROVED** [review]
LGTM, just a few improvements on the text around this change

File: doc/box2d.md (hunk 3 above)
Comment (spydon) [reply doc/box2d.md:7]: Good idea, updated it now

**Review by erickzanardo — COMMENTED** [review]
LGTM.

But I think this should be pointing to v1 right? vector 2d unification has been merged already right?

**Comment by spydon** [comment]
> LGTM.
> 
> But I think this should be pointing to v1 right? vector 2d unification has been merged already right?

Indeed, good catch! Updated.

**Pull request merged**